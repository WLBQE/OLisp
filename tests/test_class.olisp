(class foo
  (member (int a))
  (member (double b))
  (member (string c))
  (constructor a b c))

(class bar
  (member (int a))
  (member (foo f))
  (constructor f a))

(class blah
  (member (bool a))
  (member (bar b))
  (constructor a b))

(define (foo a) (foo 1 1.5 "hello"))
(define (bar b) (bar a 3))
(define (blah c) (blah true b))

(print a.a)
(print a.b)
(print a.c)
(print b.a)
(print b.f.c)
(print c.a)
(print c.b.a)
(print c.b.f.b)
