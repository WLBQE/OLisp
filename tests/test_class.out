1
1.5
hello
3
hello
true
3
1.5
